language/move-lang/src/parser/syntax.rs

```diff
@@ -1415,22 +1415,55 @@ fn parse_address_block<'input>(
 //**************************************************************************************************
 
 // Parse a use declaration:
-//      UseDecl = "use" <ModuleIdent> ("as" <ModuleName>)? ";"
-fn parse_use_decl<'input>(
-    tokens: &mut Lexer<'input>,
-) -> Result<(ModuleIdent, Option<ModuleName>), Error> {
+//      UseDecl =
+//          "use" <ModuleIdent> <UseAlias> ";" |
+//          "use" <ModuleIdent> :: <UseMember> ";" |
+//          "use" <ModuleIdent> :: "{" Comma<UseMember> "}" ";"
+fn parse_use_decl<'input>(tokens: &mut Lexer<'input>) -> Result<Use, Error> {
     consume_token(tokens, Tok::Use)?;
     let ident = parse_module_ident(tokens)?;
-    let alias = if tokens.peek() == Tok::As {
+    let alias_opt = parse_use_alias(tokens)?;
+    let use_ = match (&alias_opt, tokens.peek()) {
+        (None, Tok::ColonColon) => {
+            consume_token(tokens, Tok::ColonColon)?;
+            let sub_uses = match tokens.peek() {
+                Tok::LBrace => parse_comma_list(
+                    tokens,
+                    Tok::LBrace,
+                    Tok::RBrace,
+                    parse_use_member,
+                    "a module member alias",
+                )?,
+                _ => vec![parse_use_member(tokens)?],
+            };
+            Use::Members(ident, sub_uses)
+        }
+        _ => Use::Module(ident, alias_opt.map(ModuleName)),
+    };
+    consume_token(tokens, Tok::Semicolon)?;
+    Ok(use_)
+}
+
+// Parse an alias for a module member:
+//      UseMember = <Name> <UseAlias>
+fn parse_use_member<'input>(tokens: &mut Lexer<'input>) -> Result<(Name, Option<Name>), Error> {
+    let member = parse_identifier(tokens)?;
+    let alias_opt = parse_use_alias(tokens)?;
+    Ok((member, alias_opt))
+}
+
+// Parse an 'as' use alias:
+//      UseAs = ("as" <Name>)?
+fn parse_use_alias<'input>(tokens: &mut Lexer<'input>) -> Result<Option<Name>, Error> {
+    Ok(if tokens.peek() == Tok::As {
         tokens.advance()?;
-        Some(parse_module_name(tokens)?)
+        Some(parse_identifier(tokens)?)
     } else {
         None
-    };
-    consume_token(tokens, Tok::Semicolon)?;
-    Ok((ident, alias))
+    })
 }
 
+// TODO rework parsing modifiers
```

Comment:
Care to explain? What needs to be reworked? Offhand it seems like this should be a closure defined inside parse_module, but otherwise, it seems reasonable to me.

Reply:
It's because of the error messages. If you do something like
```
module M {
  foo = 5;
}
```
You get an error that says "expected 'fun'". I would like something that says "expected module member: `fun`, `use`, `struct`, or `const`"

This requires changing how we handle modifiers